reader.go

```diff
@@ -1039,6 +1056,14 @@ func (config *ReaderConfig) Validate() error {
 		return errors.New(fmt.Sprintf("minimum batch size greater than the maximum (min = %d, max = %d)", config.MinBytes, config.MaxBytes))
 	}
 
+	if config.ReadBackoffMax < 0 || config.ReadBackoffMax < config.ReadBackoffMin {
+		return errors.New(fmt.Sprintf("ReaxBackoffMax out of bounds: %d", config.ReadBackoffMax))
```

Comment:
```suggestion
		return errors.New(fmt.Sprintf("ReadBackoffMax out of bounds: %d", config.ReadBackoffMax))
```

What do you think of breaking this out into two separate checks and errors?   As an end user, I think that it would be easier to debug the issue where max < min.

I also think that it might be useful to do the max < min check *after* applying default values.  As currently written, I think this would error if I wanted to use the default for `ReadBackoffMax` but specified a non-zero value for `ReadBackoffMin`.